src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/ProgramService.kt

```diff
@@ -0,0 +1,104 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.ProgramService
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ProgramServiceRepository
+
+@DslMarker
+annotation class ProgramServiceDslMarker
+
+@TestDataDslMarker
+interface ProgramServiceDsl {
+  @CourseActivityDslMarker
+  fun courseActivity(
+    courseActivityId: Long = 0,
+    code: String = "CA",
+    programId: Long = 20,
+    prisonId: String = "LEI",
+    description: String = "test course activity",
+    capacity: Int = 23,
+    active: Boolean = true,
+    startDate: String = "2022-10-31",
+    endDate: String? = null,
+    minimumIncentiveLevelCode: String = "STD",
+    internalLocationId: Long? = -8,
+    excludeBankHolidays: Boolean = false,
+    dsl: CourseActivityDsl.() -> Unit = {
+      courseSchedule()
+      courseScheduleRule()
+      payRate()
+    },
+  ): CourseActivity
+}
+
+@Component
+class ProgramServiceBuilderRepository(private val programServiceRepository: ProgramServiceRepository) {
+  fun save(programService: ProgramService) = programServiceRepository.save(programService)
+}
+
+@Component
+class ProgramServiceBuilderFactory(
+  private val repository: ProgramServiceBuilderRepository? = null,
+  private val courseActivityBuilderFactory: CourseActivityBuilderFactory = CourseActivityBuilderFactory(),
+) {
+  fun builder() = ProgramServiceBuilder(repository, courseActivityBuilderFactory)
+}
```

Comment:
By getting a builder from the BuilderFactory the builder has a reference to a repository which knows how to save the ProgramService.